runtime/oti/ObjectHash.hpp

```diff
@@ -203,14 +203,89 @@ class VM_ObjectHash
 		return inlineConvertValueToHash(vm, (UDATA)objectPointer);
 	}
 
+	/**
+	 * Fetch objectPointer's hashcode when objectPointer is a contiguous object.
+	 *
+	 * @pre objectPointer must be a valid object reference
+	 *
+	 * @param vm			a java VM
+	 * @param objectPointer 	a valid object reference
+	 * @param objectClass 	class of objectPointer
+	 * @param offset 	objectPointer size offset
+	 */
+	static VMINLINE I_32
+	inlineContiguousObjectHashCode(J9JavaVM *vm, j9object_t objectPointer, J9Class *objectClass, UDATA offset)
```

Comment:
Minor: But I think indexable should be in the name of this since its assumed that `objectClass` is an array class